src/overlays/actors/ovl_Bg_Fire_Wall/z_bg_fire_wall.c

```diff
@@ -32,41 +38,184 @@ const ActorInit Bg_Fire_Wall_InitVars = {
+    {
+        ELEMTYPE_UNK0,
+        { 0x20000000, 0x01, 0x04 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NONE,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 34, 85, 0, { 0, 0, 0 } },
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_809ACC8C = { 1, 80, 100, MASS_IMMOVABLE };
-
-#endif
-
-extern ColliderCylinderInit D_809ACC60;
-extern CollisionCheckInfoInit D_809ACC8C;
-
-extern UNK_TYPE D_06000040;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/BgFireWall_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/BgFireWall_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC5C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC638.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC68C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC6C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC760.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC7F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809AC970.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/BgFireWall_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Fire_Wall/func_809ACB28.s")
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 80, 100, MASS_IMMOVABLE };
+
+static TexturePtr sFlameTextures[] = { gFwallFireball0Tex, gFwallFireball1Tex, gFwallFireball2Tex, gFwallFireball3Tex,
+                                       gFwallFireball4Tex, gFwallFireball5Tex, gFwallFireball6Tex, gFwallFireball7Tex };
+
+void BgFireWall_Init(Actor* thisx, PlayState* play) {
+    BgFireWall* this = (BgFireWall*)thisx;
+
+    this->unk_14C = this->actor.params;
+    this->actor.scale.y = 0.005f;
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    this->actor.scale.z = 0.12f;
+    this->actor.scale.x = 0.12f;
+    this->unk_15C = 0.09f;
+    this->unk_158 = 0.1f;
+    this->unk_160 = 300.0f;
+    this->texIndex = Rand_S16Offset(0, 7);
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->actionFunc = func_809AC638;
+}
+
+void BgFireWall_Destroy(Actor* thisx, PlayState* play) {
+    BgFireWall* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+s32 func_809AC5C0(BgFireWall* thisx, PlayState* play) {
+    BgFireWall* this = THIS;
+    Actor* player = play->actorCtx.actorLists[2].first;
+    Vec3f sp1C;
+
+    Actor_OffsetOfPointInActorCoords(&this->actor, &sp1C, &player->world.pos);
+    if ((fabsf(sp1C.x) < this->unk_160) && (fabsf(sp1C.z) < (this->unk_160 + 20.0f))) {
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
Personal preference. Up to you

```suggestion
    if ((fabsf(sp1C.x) < this->unk_160) && (fabsf(sp1C.z) < (this->unk_160 + 20.0f))) {
        return true;
    }
    return false;
```